Modules/pyexpat.c

```diff
@@ -1816,25 +1739,45 @@ MODULE_INITFUNC(void)
 
     if (PyModule_AddStringConstant(errors_module, "__doc__",
                                    "Constants used to describe "
-                                   "error conditions.") < 0)
-        return NULL;
+                                   "error conditions.") < 0) {
+        goto error;
+    }
 
-    if (PyModule_AddObject(errors_module, "codes", codes_dict) < 0)
-        return NULL;
-    if (PyModule_AddObject(errors_module, "messages", rev_codes_dict) < 0)
-        return NULL;
+    if (PyModule_AddObject(errors_module, "codes", codes_dict) < 0) {
+        goto error;
+    }
+    if (PyModule_AddObject(errors_module, "messages", rev_codes_dict) < 0) {
```

Comment:
Hum. Reference counting is hard. Py_XDECREF(codes_dict) on goto error is wrong.

I suggest to call Py_INCREF() before and add Py_DECREF() before return 0, for codecs_dict and rev_codecs_dict.

Reply:
@vstinner decref the dicts is still needed in goto error if the error occurs before these calls to add objects.  I attempted to fix it, see latest patch